parsers/mainz.py

```diff
@@ -0,0 +1,177 @@
+  '3' : 'cafe-rewi',
+  '4' : 'mensa-bingen',
+  '5' : 'mensa-K3',
+  '6' : 'mensa-holzstraße',
+  '7' : 'mensarium',
+  '8' : 'cafe-bingen-rochusberg',
+  '9' : 'mensablitz'
+}
+
+display = {
+  '2' : 'Aktuelle Woche',
+  '3' : 'Nächste Woche'
+}
+
+roles = ('student', 'other', 'employee')
+
+extraLegend = {
+    # Source: https://www.studierendenwerk-mainz.de/essentrinken/speiseplan/
+    '1': 'mit Farbstoff',
+    '2': 'mit Konservierungsstoff',
+    '3': 'mit Antioxidationsmittel',
+    '4': 'mit Geschmacksverstärker',
+    '5': 'geschwefelt',
+    '6': 'geschwärzt',
+    '7': 'gewachst',
+    '8': 'Phosphat',
+    '9': 'mit Süßungsmitteln',
+    '10': 'enthält eine Phenylalaninquelle',
+    'S' : 'Schweinefleisch',
+    'G' : 'Geflügelfleisch',
+    'R' : 'Rindfleisch',
+    'Gl' : 'Gluten',
+    'We' : 'Weizen (inkl. Dinkel)',
+    'Ro' : 'Roggen',
+    'Ge' : 'Gerste',
+    'Haf': 'Hafer',
+    'Kr' : 'Krebstiere und Krebstiererzeugnisse',
+    'Ei' : 'Eier und Eiererzeugnisse',
+    'Fi' : 'Fisch und Fischerzeugnisse',
+    'En' : 'Erdnüsse und Erdnusserzeugnisse',
+    'So' : 'Soja und Sojaerzeugnisse',
+    'La' : 'Milch und Milcherzeugnisse',
+    'Sl' : 'Sellerie und Sellerieerzeugnisse',
+    'Sf' : 'Senf und Senferzeugnisse',
+    'Se' : 'Sesamsamen und Sesamsamenerzeugnisse',
+    'Sw' : 'Schwefeldioxid und Sulfite > 10mg/kg',
+    'Lu' : 'Lupine und Lupinerzeugnisse',
+    'Wt' : 'Weichtiere und Weichtiererzeugnisse',
+    'Nu' : 'Schalenfrüchte',
+    'Man': 'Mandel',
+    'Has': 'Haselnüsse',
+    'Wa' : 'Walnüsse',
+    'Ka' : 'Kaschunüsse',
+    'Pe' : 'Pecanüsse',
+    'Pa' : 'Paranüsse',
+    'Pi' : 'Pistatien',
+    'Mac': 'Macadamianüsse',
+    'icon:S.png' : 'Scheinefleisch',
+    'icon:R.png' : 'Rindfleisch',
+    'icon:Fi.png' : 'Fisch',
+    'icon:Gl.png' : 'Glutenfrei',
+    'icon:La.png' : 'Lactosefrei',
+    'icon:Vegan.png' : 'Vegan',
+    'icon:Veggi.png' : 'Vegetarisch'
+    
+}
+
+def build_meal_name(meal):
+  # There are the extras of the meal inside the meal name
+  # This will remove the extras and the unnecessary spaces
+  # Example: 6 gebackene Fischstäbchen (Gl,Fi,We) mit Reis und veganem Joghurt-Kräuter-Dip (3,Gl,So,Sf,Ge)
+  # Output: 6 gebackene Fischstäbchen mit Reis und veganem Joghurt-Kräuter-Dip
+	name = ' '.join(re.sub(r'\((\w+,?)+\)', '', str(meal)).split())
+  
+  # Shorten the meal name to 250 characters like the api specification: https://doc.openmensa.org/feed/v2/#name
+	if len(name) > 250:
+			name = name[:245] + '...' 
+	
+	return name
+	
+def build_meal_notes(meal):
+  meal_name = str(meal.find('div', class_="speiseplanname").string).strip()
+  images = meal.find_all('img')
+
+  # Use a set for easy elimination of duplicates
+  notes = set()
+      
+  # extracting the icons with spezial informations about the meal
+  # Example: <img src="/fileadmin/templates/images/speiseplan/Veggi.png"/>
+  for icon in images:
+    if "icon:"+path.basename(icon['src']) in extraLegend:
```

Comment:
If you choose to split the icons into an extra dict, this could be simplified to
```
iconName = path.basename(icon['src'])
if iconName in iconLegend:
    notes.add(iconsLegend[iconName])
```